test/distributed/optim/test_zero_redundancy_optimizer.py

```diff
@@ -367,52 +335,52 @@ def some_trainable():
         all_trainable()
         some_trainable()
 
+    @skip_if_no_gpu
     def test_collect_shards(self):
         """ Check the state consolidation mechanism, and the state dict exposed by ZeroRedundancyOptimizer"""
         self.dist_init(self.rank)
         RECIPIENT_RANK = 0
 
-        # Run a dummy step so that the optimizer state dict exists
-        batch, input_width, hidden, target_width = 3, 20, 10, 5
-        target = torch.rand((batch, target_width), device=self.device)
-        inputs = torch.rand((batch, input_width), device=self.device)
+        with torch.cuda.device(self.device):
```

Comment:
trying to make sure that broadcast_object_list materializes onto the right device (cc @rohan-varma )